cluster_library.c

```diff
@@ -155,11 +157,13 @@ cluster_multibulk_resp_recursive(RedisSock *sock, size_t elements,
                 }
                 break;
             case TYPE_MULTIBULK:
-                r->element = ecalloc(r->len,sizeof(clusterReply*));
-                r->elements = r->len;
-                cluster_multibulk_resp_recursive(sock, r->elements, r->element,
-                    err TSRMLS_CC);
-                if (*err) return;
+                if (r->len >= 0) {
+                    r->element = ecalloc(r->len,sizeof(clusterReply*));
```

Comment:
What if `r->len` == 0?

Reply:
Zero-length multi-bulk is valid but I think `calloc(0, size)` is undefined behavior.  I'll not allocate the array in case `r->len==0`